### ISSUE TITLE
Cursor at wrong position, when navigating to end of line inside bold or italic text

### ISSUE THREAD

**Issue by murermader** [typora/typora-issues#5749]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

   If it only exists in 3rd party themes or css, you can still report it, but please attach the theme target or the css file. We may not "fix" it, if it is caused by 3rd party themes or css styles that we cannot support.

- [x] Searched http://support.typora.io/

----

**Describe the bug**

There are multiple effects of this bug, which I think is due to the shifting of the words when navigating over bold / italics text. Here is what happens:
- When you navigate over words that are bold, they get expanded to \*\*bold word\*\*. Navigating with the arrow key right will show the cursor briefly at the wrong location. **This is only visual!** Slightly annoying, but not too distracting
- When inside a bold word, and then navigating to the end of the line with `CMD` + `RIGHT ARROW`, the cursor ends up at the wrong location! **This is not only visual, becuase the cursor stays in the wrong location.**

**To Reproduce**
1. Write text with bold / italic words. Go inside a word, so that you can the \* characters.
2. Navigate to the end of the line
3. After navigation, the text wraps (because the \* will be removed) 
4. The cursor is now in the wrong location

**Expected behavior**
The cursor should be at the end of the line while \* are shown (in my video that would be behind 'when'), or it should be at the end of the line when \* are not shown (in my video that would be behind 'navigating')

**Screenshots / Screencasts**
I recorded a video of it here: https://share.cleanshot.com/dBX1pbG6. Skip to ~30 seconds for the bug, the other stuff is just visual bugs.

**Sample Markdown File**

```
the *cursor* moves **too** many characters, when navigating with arrows keys over italics or bold.
```

**Desktop (please complete the following information):**
 - OS: macOS Ventura 13.1.1 (a)

**Typora Version**
e.g: 1.6.5 (6860)

**Comment by rkttu** [comment]
I agree. This issue causes me to waste a lot of work time trying to position the cursor when editing markdown files.
